services/QuillLMS/app/controllers/translated_texts_controller.rb

```diff
@@ -5,12 +5,14 @@ class TranslatedTextsController < ApplicationController
 
   def index
     @translated_texts = TranslatedText
-      .all
       .includes(:english_text, :translation_mappings)
+      .order(created_at: :asc)
       .page(params[:page])
       .per(200)
 
+    @translated_texts = @translated_texts.where(locale: params[:locale]) if params[:locale].present?
     @js_file = "entrypoints/snippets/translated_text.ts"
+    @locales = TranslatedText.select(:locale).distinct.pluck(:locale)
```

Comment:
Optional: depending on the anticipated traffic on this page, consider caching this value in redis or declaring it inline statically, as it should eventually converge on a stable array value

Reply:
I think the traffic will be quite low since it's only a staff page, but you're right that it could get a little slow as we add more languages. Maybe cross that bridge if folks complain?

Reply:
I don't think we'll get beyond a couple hundred thousand records in `translated_text` and I think Postgres has this stuff pretty optimized so I doubt it'll be an issue.